Fortran/gfortran/utils/update-test-config.py

```diff
@@ -163,6 +163,20 @@ def get_lines(filepath: str) -> list[str]:
     finally:
         return lines
 
+# Check if a test configuration should be generated for the given directory
+#
+# Multi-file tests consist of a single "main" file and some number of dependent
+# sources. Some directories only contain "dependent" sources. A test
+# configuration should not be generated for such directories.
+def should_generate_config(d: str) -> bool:
+    # The directories containing only dependent sources. These should be
+    # relative to the root directory containing the gfortran tests.
+    skip: list[str] = ['regression/coarray/add_sources']
```

Comment:
This won't work on Windows (which separates names with `\`). I think instead of a string you could represent with a type intended to handle OS differences in paths? I'm not that familiar with Python but some searching found https://docs.python.org/3/library/pathlib.html#module-pathlib